### ISSUE TITLE
Install on trusty throws gpg error (spurious?)

### ISSUE THREAD

**Issue by dpb1** [CanonicalLtd/ubuntu-advantage-client#801]
Maybe related to #776, 

I get the following error on install on a fresh trusty system as the root user.  This also happens as the ubuntu user:

gpg: keyblock resource `/root/.gnupg/secring.gpg': file open error

I don't have a .gnupg dir there (is that a requirement?).

```
root@enough-parrot:~# sudo apt dist-upgrade                              
Reading package lists... Done                                            
Building dependency tree                                           
Reading state information... Done                               
Calculating upgrade... Done                                                    
The following packages were automatically installed and are no longer required:
  libfreetype6 os-prober                                                       
Use 'apt-get autoremove' to remove them.                                       
The following NEW packages will be installed:                           
  python3-pkg-resources python3-yaml                                
The following packages will be upgraded:                         
  ubuntu-advantage-tools                                          
1 upgraded, 2 newly installed, 0 to remove and 0 not upgraded.           
Need to get 185 kB of archives.                                       
After this operation, 809 kB of additional disk space will be used.     
Do you want to continue? [Y/n] y                                  
Get:1 http://archive.ubuntu.com/ubuntu/ trusty-updates/main python3-yaml amd64 3.10-4ubuntu0.1 [92.7 kB]
Get:2 http://ppa.launchpad.net/canonical-server/ua-client-daily/ubuntu/ trusty/main ubuntu-advantage-tools amd64 19.6~626~git.6200034~ubuntu14.04.1 [60.3 kB]
Get:3 http://archive.ubuntu.com/ubuntu/ trusty-updates/main python3-pkg-resources all 3.3-1ubuntu2 [31.7 kB]
Fetched 185 kB in 1s (165 kB/s)                                                
Selecting previously unselected package python3-yaml.
(Reading database ... 25111 files and directories currently installed.)
Preparing to unpack .../python3-yaml_3.10-4ubuntu0.1_amd64.deb ...
Unpacking python3-yaml (3.10-4ubuntu0.1) ...                                   
Selecting previously unselected package python3-pkg-resources.
Preparing to unpack .../python3-pkg-resources_3.3-1ubuntu2_all.deb ...
Unpacking python3-pkg-resources (3.3-1ubuntu2) ...          
Preparing to unpack .../ubuntu-advantage-tools_19.6~626~git.6200034~ubuntu14.04.1_amd64.deb ...
Moving obsolete conffile /etc/update-motd.d/99-esm out of the way...
Unpacking ubuntu-advantage-tools (19.6~626~git.6200034~ubuntu14.04.1) over (10ubuntu0.14.04.3) ...
Processing triggers for man-db (2.6.7.1-1ubuntu1) ...         
Setting up python3-yaml (3.10-4ubuntu0.1) ...               
Setting up python3-pkg-resources (3.3-1ubuntu2) ...
Setting up ubuntu-advantage-tools (19.6~626~git.6200034~ubuntu14.04.1) ...
gpg: keyblock resource `/root/.gnupg/secring.gpg': file open error
Removing obsolete conffile /etc/update-motd.d/99-esm ...
root@enough-parrot:~#
```

**Comment by dpb1** [comment]
I'm not sure if related, but my attaching after this does not work:

```
17:02:25           dpb | root@enough-parrot:~# sudo ua attach C1KRD7q2nBgvaRhghdD43rMtSWjAh
17:02:25           dpb | Failed to attach machine. See https://ubuntu.com/advantage
17:02:25           dpb | root@enough-parrot:~# 

```

**Comment by panlinux** [comment]
I just reproduced that:
```
...
Preparing to unpack .../ubuntu-advantage-tools_19.6~626~git.6200034~ubuntu14.04.1_amd64.deb ...
Moving obsolete conffile /etc/update-motd.d/99-esm out of the way...
Unpacking ubuntu-advantage-tools (19.6~626~git.6200034~ubuntu14.04.1) over (10ubuntu0.14.04.4) ...
Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
Setting up python3-yaml (3.10-4ubuntu0.1) ...
Setting up python3-pkg-resources (3.3-1ubuntu2) ...
Setting up ubuntu-advantage-tools (19.6~626~git.6200034~ubuntu14.04.1) ...
gpg: keyblock resource `/root/.gnupg/secring.gpg': file open error
Removing obsolete conffile /etc/update-motd.d/99-esm ...
```
It's the gpg call in postinst:
```
root@trusty-ua:~# gpg --no-auto-check-trustdb --no-options --no-default-keyring --keyring /usr/share/keyrings/ubuntu-advantage-keyring.gpg --export 56F7650A24C9E9ECF87C4D8D4067E40313CB4B13 --yes > foo.gpg
gpg: keyblock resource `/root/.gnupg/secring.gpg': file open error
root@trusty-ua:~# l foo.gpg 
-rw-r--r-- 1 root root 2.3K Oct  7 20:45 foo.gpg
root@trusty-ua:~# 
```

**Comment by panlinux** [comment]
Attaching worked fine, though:
```
root@trusty-ua:~# ua attach lolsecret
Enabling default service esm-infra
Updating package lists
ESM Infra enabled.
This machine is now attached to 'andreas.hasenack@canonical.com'.

SERVICE       ENTITLED  STATUS    DESCRIPTION
cc-eal        no        —         Common Criteria EAL2 Provisioning Packages
cis-audit     no        —         Center for Internet Security Audit Tools
esm-infra     yes       enabled   UA Infra: Extended Security Maintenance
fips          no        —         NIST-certified FIPS modules
fips-updates  no        —         Uncertified security updates to FIPS modules
livepatch     yes       n/a       Canonical Livepatch service

Enable services with: ua enable <service>

     Account: andreas.hasenack@canonical.com
Subscription: andreas.hasenack@canonical.com
```
This is a lxd container, hence the `n/a` for livepatch.

**Comment by blackboxsw** [comment]
This is addressed by PR #803. No dependency on gpg cli tooling, means no more error message here.

**Comment by powersj** [comment]
0bc4351eccb63a07082abae9d9ad6d168e2c004a fixes this. Marking closed.
